pkg/auth/providers/common/ldap/ldap_util.go

```diff
@@ -211,6 +226,75 @@ func AttributesToPrincipal(attribs []*ldapv2.EntryAttribute, dnStr, scope, provi
 	return principal, nil
 }
 
+func GatherParentGroups(groupPrincipal v3.Principal, searchDomain string, groupScope string, config *ConfigAttributes, lConn *ldapv2.Conn,
+	groupMap map[string]bool, nestedGroupPrincipals *[]v3.Principal) error {
+	groupMap[groupPrincipal.ObjectMeta.Name] = true
+	principals := []v3.Principal{}
+	var searchAttributes []string
+	parts := strings.SplitN(groupPrincipal.ObjectMeta.Name, ":", 2)
+	if len(parts) != 2 {
+		return errors.Errorf("invalid id %v", groupPrincipal.ObjectMeta.Name)
+	}
+	groupDN := strings.TrimPrefix(parts[1], "//")
+
+	if config.ProviderName == "activedirectory" {
+		searchAttributes = []string{config.MemberOfAttribute, config.ObjectClass, config.GroupObjectClass, config.UserLoginAttribute, config.GroupNameAttribute,
+			config.GroupSearchAttribute}
+	} else if config.ProviderName == "openldap" {
+		searchAttributes = []string{config.GroupMemberUserAttribute, config.GroupMemberMappingAttribute, config.ObjectClass, config.GroupObjectClass, config.UserLoginAttribute,
+			config.GroupNameAttribute, config.GroupSearchAttribute}
+	}
```

Comment:
searchAttributes is a constant array of attributes, this can be passed as a parameter to this method - instead of evaluating this if-else in each recursive call